consensus/ibft/src/main/java/org/hyperledger/besu/consensus/ibft/statemachine/IbftBlockHeightManager.java

```diff
@@ -133,6 +137,30 @@ public void handleBlockTimerExpiry(final ConsensusRoundIdentifier roundIdentifie
     }
   }
 
+  /**
+   * If the list of validators for the next block to be proposed/imported has changed from the
+   * previous block, log the change. Only log for round 0 (i.e. once per block).
+   *
+   * @param ibftRound The current round
+   */
+  private void logValidatorChanges(final IbftRound ibftRound) {
+    if (ibftRound.getRoundIdentifier().getRoundNumber() == 0) {
+      final Collection<Address> previousValidators =
+          MessageValidatorFactory.getValidatorsForBlock(ibftRound.protocolContext, parentHeader);
+      final Collection<Address> validatorsForHeight =
+          MessageValidatorFactory.getValidatorsAfterBlock(ibftRound.protocolContext, parentHeader);
+      if (!(validatorsForHeight.containsAll(previousValidators))
```

Comment:
Are the two containsAll calls used so that the validators can be compared ignoring order?

Reply:
Not to ignore order, but to check for both new-validator and removed-validator cases.